narwhals/_pandas_like/series.py

```diff
@@ -851,6 +851,11 @@ def rolling_mean(
     def __iter__(self: Self) -> Iterator[Any]:
         yield from self._native_series.__iter__()
 
+    def __contains__(self: Self, other: Any) -> bool:
+        if other is None:
+            return self._native_series.isna().any()  # type: ignore[no-any-return]
+        return other in self._native_series
```

Comment:
I fell for the example contained in our [why](https://narwhals-dev.github.io/narwhals/why/) section!

There are two ways of doing this:

- `return other in self.to_numpy()`: yet I don't think this is a good idea for cudf
- `return self._native_series.isin({other}).any()`

Reply:
May I ask a general, possibly silly, question, just for the sake of better understanding? 😇 

I see the point in implementing `__contains__` for _pandas-like_ and _arrow_ and also what you mention here. Instead, I haven't properly got the details behind https://github.com/narwhals-dev/narwhals/issues/1443#issuecomment-2508932494, as I don't see `func(1, pa.chunked_array([data]))` returning `False`.

Should we implement `__contains__` at `narwhals/series.py` to ensure the above behaviour is explicitly triggered via `__contains__`?

Reply:
> as I don't see func(1, pa.chunked_array([data])) returning False.

Damn that's true... but now I don't know why, and I wonder what is being called. The following returns false:

```py
1 in pa.chunked_array([[1,2,3]])
False
```

> Should we implement __contains__ at narwhals/series.py to ensure the above behaviour is explicitly triggered via __contains__?

I am addressing it in the PR

Reply:
> Damn that's true... but now I don't know why, and I wonder what is being called.

Me neither, had issues in trying to figure it out

> I am addressing it in the PR

yeah, I see it; was just asking to confirm my understanding as I couldn't indeed see where this was coming from, without the explicit reference to `__contains__` 😃

Reply:
Ok, since we implement `__iter__` the check works.

Example:

```py
class A:
    pass

class B:
    def __iter__(self):
        yield from [1,2,3]

0 in B(), 1 in B()
> (False, True)

1 in A()
> TypeError: argument of type 'A' is not iterable
```

Reply:
nice, thanks for the clarification!:)